catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/GmdTransformer.java

```diff
@@ -0,0 +1,302 @@
+        crs.append(COLON);
+        crs.append(StringUtils.defaultString(pathValueTracker.getPathValue(CRS_VERSION_PATH)));
+        crs.append(COLON);
+        crs.append(StringUtils.defaultString(pathValueTracker.getPathValue(CRS_CODE_PATH)));
+        metacard.setAttribute(GmdMetacardType.GMD_CRS, crs.toString());
+
+    }
+
+    private void addMetacardTitle(final XstreamPathValueTracker pathValueTracker,
+            MetacardImpl metacard) {
+        String title = pathValueTracker.getPathValue(TITLE_PATH);
+        if (StringUtils.isNotEmpty(title)) {
+            metacard.setTitle(title);
+        }
+    }
+
+    private void addMetacardAbstract(final XstreamPathValueTracker pathValueTracker,
+            MetacardImpl metacard) {
+        String gmdAbstract = pathValueTracker.getPathValue(ABSTRACT_PATH);
+        if (StringUtils.isNotEmpty(gmdAbstract)) {
+            metacard.setAttribute(GmdMetacardType.GMD_ABSTRACT, gmdAbstract);
+        }
+
+    }
+
+    private void addMetacardFormat(final XstreamPathValueTracker pathValueTracker,
+            MetacardImpl metacard) {
+        String format = pathValueTracker.getPathValue(FORMAT_PATH);
+        if (StringUtils.isNotEmpty((format))) {
+            metacard.setAttribute(GmdMetacardType.GMD_FORMAT, format);
+        }
+    }
+
+    private void addMetacardResourceUri(final XstreamPathValueTracker pathValueTracker,
+            MetacardImpl metacard) {
+        String linkage = pathValueTracker.getPathValue(LINKAGE_URI_PATH);
+        if (StringUtils.isNotEmpty(linkage)) {
+            try {
+                metacard.setResourceURI(new URI(linkage));
+            } catch (URISyntaxException e) {
+                LOGGER.info("Unable to read resource URI {]", linkage);
+            }
+        }
+
+    }
+
+    private void addMetacardSubject(final XstreamPathValueTracker pathValueTracker,
+            MetacardImpl metacard) {
+
+        List<String> subjects = pathValueTracker.getAllValues(KEYWORD_PATH);
+        subjects.addAll(pathValueTracker.getAllValues(TOPIC_CATEGORY_PATH));
+        metacard.setAttribute(GmdMetacardType.GMD_SUBJECT, (Serializable) subjects);
+
+    }
+
+    private void addMetacardLocation(final XstreamPathValueTracker pathValueTracker,
+            MetacardImpl metacard) {
+        String westLon = pathValueTracker.getPathValue(BBOX_WEST_LON_PATH);
+        String eastLon = pathValueTracker.getPathValue(BBOX_EAST_LON_PATH);
+        String southLat = pathValueTracker.getPathValue(BBOX_SOUTH_LAT_PATH);
+        String northLat = pathValueTracker.getPathValue(BBOX_NORTH_LAT_PATH);
+
+        if (westLon != null && eastLon != null && southLat != null && northLat != null) {
+            WKTWriter wktWriter = new WKTWriter();
+
+            GeometryFactory factory = new GeometryFactory();
+            Envelope envelope = new Envelope(Double.parseDouble(eastLon),
+                    Double.parseDouble(westLon),
```

Comment:
Also, should you call trim() on the strings first, just to make sure some stray whitespace doesn't blow up the code?